cmd/manager/main.go

```diff
@@ -159,8 +159,15 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Get the global CNI options(cluster specific) from clusterNetworkConfig
+	clusterLevelCniOptions := clusterconfig.network.ClusterLevelCniOptions()
+	if err != nil {
+		log.Error(err, "failed to get cni Options from network")
+		os.Exit(1)
+	}
+
 	// Setup all Controllers
-	if err := controller.AddToManager(mgr); err != nil {
+	if err := controller.AddToManager(mgr, clusterLevelCniOptions); err != nil {
```

Comment:
You're passing the CNI options to all controllers, present and future. This seems like it will be a WMCO specific requirement, I feel like this should be done in the `newReconciler` function.